### PR TITLE
[Build] webpack5 upgrade

### PR DESCRIPTION
**Pull request by nikhilmandlik**
Closes: 
https://github.com/nasa/openmct/issues/3899
https://github.com/nasa/openmct/issues/2449
https://github.com/nasa/openmct/issues/4365
https://github.com/nasa/openmct/issues/3966

### TODOs:
* [ ] Fix test coverage issue with `karma-coverage-istanbul-reporter` - explore `babel-plugin-istanbul`and https://github.com/mattlewis92/karma-coverage-istanbul-reporter. Moved to another Ticket https://github.com/nasa/openmct/issues/4640
* [x] separate dev, prod configs #4629 
* [x] ~~Add hash for css, index and js files~~ (can be done in a new PR) (there is issue with css hash as we load themes and need absolute name of theme/css file to load)
* [x] ~~Verify that this implementation will work with e2e code coverage~~ (@johnhill2 )
* [x] ~~Attempt to build in node16 to close out this ticket https://github.com/nasa/openmct/pull/4242~~  (can be done in a new PR)
* [x] Verify that this does not break VISTA (@davetsay)
* [x] Verify that this does not break NIRVSS (@nikhilmandlik )
* [x] Verify that this does not break viper-openmct on /dev (@nikhilmandlik )

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?

### Author Checklist

* [x] Changes address original issue?
* [x] ~~Unit tests included and/or updated with changes?~~ N/A
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] ~~Testing instructions included in associated issue?~~ N/A

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -18,12 +18,16 @@ jobs:
     executor: linux
     steps:
       - checkout
-      - restore_cache:
```
Comment (unlikelyzero) [inline .circleci/config.yml:21]: we should keep the npm cache otherwise we'll dramatically increase the runtime of the tests

File: package.json [hunk 2]
```diff
@@ -64,10 +64,10 @@
     "uuid": "^3.3.3",
     "v8-compile-cache": "^1.1.0",
     "vue": "2.5.6",
-    "vue-loader": "^15.2.6",
+    "vue-loader": "^15.9.7",
     "vue-template-compiler": "2.5.6",
-    "webpack": "^4.16.2",
-    "webpack-cli": "^3.1.0",
+    "webpack": "^5.37.0",
+    "webpack-cli": "^3.3.12",
```
Comment (unlikelyzero) [inline package.json:70]: I think this is incompatible with 5.x

File: .circleci/config.yml (hunk 1 above)
Comment (nikhilmandlik) [reply .circleci/config.yml:21]: this is for testing only ll be reverting later on

**Comment by akhenry** [comment]
Blocked by lack of support for Webpack 5 by code coverage tooling

**Review by trusktr — APPROVED** [review]

**Comment by unlikelyzero** [comment]
@trusktr part of this PR was to explore our use of dev and prod build scripts. Can you review those?

**Comment by akhenry** [comment]
@davetsay to build VISTA against this branch to see if there are any issues.

**Comment by davetsay** [comment]
@nikhilmandlik , @trusktr , built in VISTA. no build or start issues. As mentioned by @trusktr , _IF_ there are any runtime issues they would be css based so i'll do a little more investigation.

Looks good to go though from a VISTA dependency stand point.

**Review by trusktr — CHANGES_REQUESTED** [review]
dev/prod scripts

**Comment by nikhilmandlik** [comment]
> * separate webpack dev prod configs. #4629

https://github.com/nasa/openmct/pull/4629

**Comment by nikhilmandlik** [comment]
> dev/prod configs

https://github.com/nasa/openmct/pull/4629

**Review by trusktr — APPROVED** [review]

**Comment by trusktr** [comment]
@unlikelyzero dev vs prod configs added by Nikhil, and tests passing. Can you double check that the tests are running as you expect? Seems like they are, but you'd know best. If so, looks like this is good to go (and a couple points in the original post can be done in new PRs).

**Comment by unlikelyzero** [comment]
@trusktr  I will make a follow-up issue for the code coverage implementation. Unfortunately, it looks like we will need to migrate to Babel/ Istanbul

**Review by unlikelyzero — APPROVED** [review]
Looks like it passes all the necessary checks. We'll need to immediately remove the legacy-peer-deps and implement babel-plugin-instanbul in a follow ticket https://github.com/nasa/openmct/issues/4640

**Pull request merged**